src/ui/functions/get-product.ts

```diff
@@ -0,0 +1,73 @@
+import { AzureFunction } from "@azure/functions";
+
+import { pipe, flow } from "fp-ts/lib/function";
+import { sequenceS } from "fp-ts/lib/Apply";
+import * as RE from "fp-ts/lib/ReaderEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+
+import { createHandler } from "@pagopa/handler-kit";
+import { jsonResponse, HttpRequest, path } from "@pagopa/handler-kit/lib/http";
+import * as azure from "@pagopa/handler-kit/lib/azure";
+import {
+  BadRequestError,
+  NotFoundError,
+} from "@pagopa/handler-kit/lib/http/errors";
+
+import { ProductDetailView } from "../api-models/ProductDetailView";
+import { requireSubscriptionId, errorResponse } from "../http";
+
+import { Product, ProductId } from "../../signature-request/product";
+import { Subscription } from "../../signature-request/subscription";
+
+import { getProduct } from "../../infra/azure/cosmos/product";
+
+export const requireProductId: (
+  req: HttpRequest
+) => E.Either<Error, Product["id"]> = flow(
+  path("productId"),
+  E.fromOption(() => new BadRequestError("Missing productId in path")),
```

Comment:
Could it be useful to define all error messages in a separate file?

Reply:
No, because the error message is tightly related to the code in which it is put and requires no localization (it will always be in English) 

If the same type of error occurs multiple times with the same type of message, I prefer to extract it as a new error (as done with `ProductNotFoundError`)